front/app/modules/commercial/content_builder/admin/components/CraftComponents/Iframe/index.tsx

```diff
@@ -0,0 +1,197 @@
+        <Input
+          label={
+            <span>
+              {formatMessage(messages.iframeUrlLabel)}{' '}
+              <IconTooltip
+                icon="info3"
+                content={formatMessage(messages.iframeUrlLabelTooltip)}
+              />
+            </span>
+          }
+          placeholder={formatMessage(messages.iframeUrlPlaceholder)}
+          type="text"
+          value={url}
+          onChange={(value) => {
+            const validation = validateUrl(value);
+            setProp((props) => (props.url = value));
+            setProp((props) => (props.errorType = validation[1]));
+            setProp((props) => (props.hasError = !validation[0]));
+            eventEmitter.emit('contentBuilderError', {
+              [id]: !validation[0],
+            });
+          }}
+        />
+        {hasError && (
+          <Error
+            marginTop="4px"
+            text={
+              errorType === 'invalidUrl'
+                ? formatMessage(messages.iframeInvalidUrlErrorMessage)
+                : formatMessage(messages.iframeWhitelistUrlErrorMessage)
+            }
+          />
+        )}
+      </Box>
+      <Box flex="0 0 100%">
+        <Input
+          label={
+            <span>
+              {formatMessage(messages.iframeHeightLabel)}{' '}
+              <IconTooltip
+                icon="info3"
+                content={formatMessage(messages.iframeHeightLabelTooltip)}
+              />
+            </span>
+          }
+          placeholder={formatMessage(messages.iframeHeightPlaceholder)}
+          type="text"
+          value={height}
+          onChange={(value) => {
+            setProp((props) => (props.height = value));
+          }}
+        />
+      </Box>
+      <Box flex="0 0 100%">
+        <Input
+          type="text"
+          id="iframeTitleInput"
+          onChange={(value) => {
+            setProp((props) => (props.title = value));
+          }}
+          value={title}
+          label={
+            <span>
+              {formatMessage(messages.iframeTitleLabel)}{' '}
+              <IconTooltip
+                icon="info3"
+                content={formatMessage(messages.iframeTitleTooltip)}
+              />
+            </span>
+          }
+        />
+      </Box>
+    </Box>
+  );
+});
+
+Iframe.craft = {
+  props: {
+    url: '',
+    height: '',
+  },
+  related: {
+    settings: IframeSettings,
+  },
+};
+
+/*
+ * Function to validate embed URL against white list
+ * Returns: boolean value whether URL input string is valid or not
+ */
+const validateUrl = (url: string) => {
+  // Used this reference for generating a valid URL regex:
+  // https://tutorial.eyehunts.com/js/url-regex-validation-javascript-example-code/
+  const validUrlCheck = url.match(
+    /(http(s)?:\/\/.)?(www\.)?[-a-zA-Z0-9@:%._+~#=]{2,256}\.[a-z]{2,6}\b([-a-zA-Z0-9@:%_+.~#?&//=]*)/g
+  );
+  if (validUrlCheck === null) {
+    return [false, 'invalidUrl'];
+  }
+
+  const urlWhiteList = [
```

Comment:
Adding this whitelist in the FE means that it can become out of sync with the list in the BE. Shouldn't this list come from the BE?

Reply:
Ah that's true.. do we have this list implemented on the BE though? I took a look but could only see the video and survey whitelists. I'll check with Sebi.

Reply:
Indeed that would be ideal, because keeping copies is a maintenance issue. But please do not let this block you. We can address the comment later. Let's get this PR merged! :-)

Reply:
Ah ok - I think Ruby and TypeScript Regex might have a slightly different format. I tried using the BE regex for the videos directly at least, and the FE complained. But sounds good - I can look into that after the [BE whitelist case](https://citizenlab.atlassian.net/browse/CL-740) is finished then :+1:

Reply:
You are correct. The format is different in both languages.